internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +285,24 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"web_application_firewall_settings": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
```

Comment:
can there be more then one web application firewall settings block? if not this should be maxitems:1